js/decklist/dci.js

```diff
@@ -0,0 +1,71 @@
+var primes = [43,47,53,71,73,31,37,41,59,61,67,29];
+function isValid(number){
+  number += "";
+  if(number.length >= 8){
+    if(number.length == 8 || number.length == 10){
+      check = getCheckDigit(number.substring(1));
+      if(number[0] == check){
+        return true;
+      }
+      else return false;
+    }
+    else return false;
+  }
+  else return -1; // for a under 8-digit number, return -1 (evals to true)
+}
+function getTenDigit(number){
+  number += "";
+  if(number.length == 10){
+    if(isValid(number)){
+      return parseInt(number);
+    }
+    else{
+      return 0;
+    }
+  }
+  else if (number.length <= 8 && isValid(number)){
+    number = getEightDigit(number);
+    number = "0" + number; 
```

Comment:
`number = "0" + getEightDigit(number)` to remove above line? I'll admit that it might be clearer as written.